weblab/static/js/db.js

```diff
@@ -526,29 +570,44 @@ function prepareMatrix ()
 	$("#comparisonLink").click(function () {
 	    document.location = $(this).data("href");
 	});
-	$("#comparisonMatrix").click(function () {
+  $("#comparisonMatrix").click(function () {
     var url = $(div).data('base-href');
     if (url.substr(-1) === '/') {
       url = url.slice(0, -1);
     }
-		if (linesToCompare.row.length > 0)
-		{
-			url += "/models";
-			for (var i=0; i<linesToCompare.row.length; i++)
-				url += "/" + $("#matrix-entry-" + linesToCompare.row[i] + "--1").data("modelId");
-		}
-		else
-			url += baseUrls.row;
-		if (linesToCompare.col.length > 0)
-		{
-			url += "/protocols";
-			for (var i=0; i<linesToCompare.col.length; i++)
-				url += "/" + $("#matrix-entry--1-" + linesToCompare.col[i]).data("protoId");
-		}
-		else
-			url += baseUrls.col;
-		document.location.href = url; // TODO: use history API instead?
-	});
+
+    if (linesToCompare.row.length > 0) {
+      var rows = linesToCompare.row.map(i => $("#matrix-entry-" + i + "--1"));
+      var modelIds = rows.map($row => $row.data('modelId'));
+      var modelVersions = rows.map($row => $row.data('modelVersion'));
+      if (components.modelVersions) {
+        url += '/models/' + modelIds[0] + '/versions/' + modelVersions.join('/');
+      } else {
+        url += '/models/' + modelIds.join('/');
+      }
+    }
+    else
+    {
+      url += baseUrls.row;
+    }
+
+    if (linesToCompare.col.length > 0)
+    {
+      var cols = linesToCompare.col.map(i => $("#matrix-entry--1-" + i));
+      var protoIds = cols.map($col => $col.data('protoId'));
+      var protoVersions = cols.map($col => $col.data('protoVersion'));
```

Comment:
There's something not quite right here - I'm getting URLs like `http://127.0.0.1:8000/experiments/models/1/4/protocols/3/versions///` generated when I try to show a submatrix!

Reply:
I also noticed when trying to 'Compare selected experiments' with only one green box selected, I get a 404, since the URL pattern requires 2 versions. In WL1 the compare view still works if only 1 experiment is selected - can we do the same?